data.md

```diff
@@ -435,29 +358,25 @@ These parsers can interperet hex-encoded strings as `Int`s, `WordStack`s, and `M
  // -------------------------------------------------------------------
     rule #alignHexString(S) => S             requires         lengthString(S) modInt 2 ==Int 0
     rule #alignHexString(S) => "0" +String S requires notBool lengthString(S) modInt 2 ==Int 0
+```
 
-    syntax WordStack ::= #parseByteStack ( String )    [function]
-                       | #parseHexBytes     ( String ) [function]
-                       | #parseHexBytesAux  ( String ) [function]
-                       | #parseHexBytesAux  ( Bytes  ) [function]
-                       | #parseByteStackRaw ( String ) [function]
-                       | #parseByteStackRaw ( String , WordStack , Int , Int ) [function, klabel(#parseByteStackRawAux)]
- // --------------------------------------------------------------------------------------------------------------------
+```k
+    syntax Bytes ::= #parseHexBytes     ( String ) [function]
+                   | #parseHexBytesAux  ( String ) [function]
+                   | #parseByteStack    ( String ) [function, memo]
+                   | #parseByteStackRaw ( String ) [function]
+ // ---------------------------------------------------------------
     rule #parseByteStack(S) => #parseHexBytes(replaceAll(S, "0x", ""))
 
-    rule #parseHexBytes(S)  => #parseHexBytesAux(#alignHexString(S))
-    rule #parseHexBytesAux("") => .WordStack
-    rule #parseHexBytesAux(S)  => #parseHexBytesAux( Int2Bytes(lengthString(S) /Int 2, String2Base(S, 16), BE) )
+    rule #parseHexBytes(S)     => #parseHexBytesAux(#alignHexString(S))
+    rule #parseHexBytesAux("") => .Bytes
+    rule #parseHexBytesAux(S)  => Int2Bytes(lengthString(S) /Int 2, String2Base(S, 16), BE)
       requires lengthString(S) >=Int 2
 
-    rule #parseHexBytesAux(BS) => BS[0] : BS[1] : BS[2] : BS[3] : #parseHexBytesAux( substrBytes(BS, 4, lengthBytes(BS)) ) requires lengthBytes(BS) >=Int 4
-    rule #parseHexBytesAux(BS) => BS[0] : BS[1] : BS[2] : .WordStack                                                       requires lengthBytes(BS) ==Int 3
-    rule #parseHexBytesAux(BS) => BS[0] : BS[1] : .WordStack                                                               requires lengthBytes(BS) ==Int 2
-    rule #parseHexBytesAux(BS) => BS[0] : .WordStack                                                                       requires lengthBytes(BS) ==Int 1
-    rule #parseHexBytesAux(BS) => .WordStack                                                                               requires lengthBytes(BS) ==Int 0
-
-    rule #parseByteStackRaw(S) => #parseHexBytesAux(String2Bytes(S))
+    rule #parseByteStackRaw(S) => String2Bytes(S)
```

Comment:
Can we re-arrange this so the rules are in the same order the syntax are declared in? Or even separate the declaration of the syntax for `#parseByteStack*` from `#parseHexBytes*`?